contracts/contracts/strategies/GeneralizedUniswapV3Strategy.sol

```diff
@@ -0,0 +1,850 @@
+                amount1Desired: desiredAmount1,
+                amount0Min: maxSlippage == 0
+                    ? 0
+                    : (desiredAmount0 * (10000 - maxSlippage)) / 10000, // Price Slippage,
+                amount1Min: maxSlippage == 0
+                    ? 0
+                    : (desiredAmount1 * (10000 - maxSlippage)) / 10000, // Price Slippage,
+                deadline: block.timestamp
+            });
+
+        (liquidity, amount0, amount1) = positionManager.increaseLiquidity(
+            params
+        );
+
+        p.liquidity += liquidity;
+
+        emit UniswapV3LiquidityAdded(p.tokenId, amount0, amount1, liquidity);
+    }
+
+    /**
+     * @notice Removes liquidity of the position in the pool
+     * @param p Position object
+     * @param liquidity Amount of liquidity to remove form the position
+     * @return amount0 Amount of token0 received after liquidation
+     * @return amount1 Amount of token1 received after liquidation
+     */
+    function _decreaseLiquidityForPosition(
+        Position storage p,
+        uint128 liquidity
+    ) internal returns (uint256 amount0, uint256 amount1) {
+        require(p.exists, "Unknown position");
+
+        (uint160 sqrtRatioX96, , , , , , ) = IUniswapV3Pool(platformAddress)
+            .slot0();
+        (uint256 exactAmount0, uint256 exactAmount1) = uniswapV3Helper
+            .getAmountsForLiquidity(
+                sqrtRatioX96,
+                p.sqrtRatioAX96,
+                p.sqrtRatioBX96,
+                liquidity
+            );
+
+        INonfungiblePositionManager.DecreaseLiquidityParams memory params = INonfungiblePositionManager
+            .DecreaseLiquidityParams({
+                tokenId: p.tokenId,
+                liquidity: liquidity,
+                amount0Min: maxSlippage == 0
+                    ? 0
+                    : (exactAmount0 * (10000 - maxSlippage)) / 10000, // Price Slippage,
+                amount1Min: maxSlippage == 0
+                    ? 0
+                    : (exactAmount1 * (10000 - maxSlippage)) / 10000, // Price Slippage,
+                deadline: block.timestamp
+            });
+
+        (amount0, amount1) = positionManager.decreaseLiquidity(params);
+
+        p.liquidity -= liquidity;
+
+        emit UniswapV3LiquidityRemoved(p.tokenId, amount0, amount1, liquidity);
+    }
+
+    /***************************************
+            ERC721 management
+    ****************************************/
+
+    /// Callback function for whenever a NFT is transferred to this contract
+    // solhint-disable-next-line max-line-length
+    /// Ref: https://docs.openzeppelin.com/contracts/3.x/api/token/erc721#IERC721Receiver-onERC721Received-address-address-uint256-bytes-
+    function onERC721Received(
+        address,
+        address,
+        uint256,
+        bytes calldata
+    ) external returns (bytes4) {
+        // TODO: Should we reject unwanted NFTs being transfered to the strategy?
```

Comment:
is there any downside to leaving unexpected NFTs on the contract?

Reply:
None as far as I can tell, but just wanted to know if anyone else thinks otherwise